src/scry/context.cr

```diff
@@ -61,18 +61,30 @@ module Scry
       case msg.method
       when "textDocument/hover"
         text_document = TextDocument.new(params, msg.id)
+        if text_document.in_memory?
+          Log.logger.debug("Ignoring path: #{text_document.filename}")
+          return Protocol::ResponseMessage.new(msg.id, nil)
+        end
         hover = HoverProvider.new(@workspace, text_document)
         response = hover.run
         Log.logger.debug(response)
         response
       when "textDocument/definition"
         text_document = TextDocument.new(params, msg.id)
+        if text_document.in_memory?
+          Log.logger.debug("Ignoring path: #{text_document.filename}")
+          return Protocol::ResponseMessage.new(msg.id, nil)
+        end
         definitions = Implementations.new(@workspace, text_document)
         response = definitions.run
         Log.logger.debug(response)
         response
       when "textDocument/completion"
         text_document, method_db = @workspace.get_file(TextDocument.uri_to_filename(params.text_document.uri))
+        if text_document.in_memory?
+          Log.logger.debug("Ignoring path: #{text_document.filename}")
+          return Protocol::ResponseMessage.new(msg.id, nil)
+        end
```

Comment:
Could you write a helper method to avoid this kind of repetition?

So we can do something like:
```cr
return ignore_path_response(msg.id, text_document) if text_document.in_memory?
```
For the `dispatch_notification(...)` you can simply return `nil` (no helper), there is no need to return a response for a notification

Reply:
Can do! Thanks!